src/main/java/core/basesyntax/figure/producers/FigureProducer.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax.figure.producers;
+
+import core.basesyntax.figure.Figure;
+import core.basesyntax.figure.figures.Circle;
+import core.basesyntax.figure.figures.IsoscelesTrapezoid;
+import core.basesyntax.figure.figures.Rectangle;
+import core.basesyntax.figure.figures.RightTriangle;
+import core.basesyntax.figure.figures.Square;
+import java.util.Random;
+
+public class FigureProducer {
+
+    public static Figure getRandomFigure() {
+        Random randomizer = new Random();
+        int bound = 1000;
+
+        switch (new Random().nextInt(5)) {
+            case 0:
+                return new Circle(ColorProducer.getRandomColor(),
+                        randomizer.nextInt(bound));
+            case 1:
+                return new Square(ColorProducer.getRandomColor(),
+                        randomizer.nextInt(bound));
+            case 2:
+                return new Rectangle(ColorProducer.getRandomColor(),
+                        randomizer.nextInt(bound),
+                        randomizer.nextInt(bound));
+            case 3:
+                return new RightTriangle(ColorProducer.getRandomColor(),
+                        randomizer.nextInt(bound),
+                        randomizer.nextInt(bound));
+            case 4:
+            default:
+                return new IsoscelesTrapezoid(ColorProducer.getRandomColor(),
+                        randomizer.nextInt(bound),
+                        randomizer.nextInt(bound),
+                        randomizer.nextInt(bound));
+        }
+    }
+
+    public static Figure[] getRandomFigures() {
```

Comment:
we have duplicated code here, you can just hardcode amount value and call your getRandomFigures(int amount) method from here

Reply:
Nice tip, thx